File: segundos 4 lenguajes/Java/JugadorVIP.java
package Java;
public class JugadorVIP {
    String nombre;
    int nivel;

    public JugadorVIP() {
        nombre = "VIP Juan";
        nivel = 50;
    }

    public JugadorVIP(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorVIP(String nombre) {
        this.nombre = nombre;
        this.nivel = 99;
    }

    public JugadorVIP(JugadorVIP j) {
        this.nombre = j.nombre;
        this.nivel = j.nivel;
    }
}

File: segundos 4 lenguajes/go/JugadorVIP.go
package main

type JugadorVIP struct {
	Nombre string
	Nivel  int
}

func NewJugadorVIP() JugadorVIP {
	return JugadorVIP{"VIP Juan", 50}
}

func NewJugadorVIPParams(nombre string, nivel int) JugadorVIP {
	return JugadorVIP{nombre, nivel}
}

func NewJugadorVIPNombre(nombre string) JugadorVIP {
	return JugadorVIP{nombre, 99}
}

func NewJugadorVIPCopy(j JugadorVIP) JugadorVIP {
	return JugadorVIP{j.Nombre, j.Nivel}
}
